### PR TITLE
fix(memory): use checked arithmetic in memfd_backed region sum

### PR DESCRIPTION
**Pull request by dan1d**
## Summary

The `memfd_backed` function uses an unchecked `.sum()` to calculate the total size of memory regions:

```rust
let size = regions.iter().map(|&(_, size)| size as u64).sum();
```

This could theoretically overflow if given sufficiently large region sizes.

This PR applies the same fix that was made to `snapshot_file` in commit 837c2e782: replace the unchecked sum with `try_fold` and `checked_add`, returning `MemoryError::OffsetTooLarge` on overflow.

## Changes

- Replace `.sum()` with `.try_fold()` + `checked_add()` in `memfd_backed`
- Return `MemoryError::OffsetTooLarge` on overflow (consistent with `snapshot_file`)

## Test plan

- [ ] Existing unit tests pass
- [ ] The fix mirrors the pattern already used in `snapshot_file` (lines 909-912)

## Related

- Commit 837c2e782 fixed the same pattern in `snapshot_file`
- This ensures consistency across memory allocation functions

### CODE REVIEW

File: src/vmm/src/vstate/memory.rs [hunk 1]
```diff
@@ -870,7 +870,10 @@ pub fn memfd_backed(
     track_dirty_pages: bool,
     huge_pages: HugePageConfig,
 ) -> Result<Vec<GuestRegionMmap>, MemoryError> {
-    let size = regions.iter().map(|&(_, size)| size as u64).sum();
+    let size = regions
+        .iter()
+        .try_fold(0u64, |acc, &(_, size)| acc.checked_add(size as u64))
+        .ok_or(MemoryError::OffsetTooLarge)?;
```
Comment (zulinx86) [inline src/vmm/src/vstate/memory.rs:922]: nice-to-have: Could we add a small unit test for the overflow case? For example,

```rs
let regions = [
    (GuestAddress(0), usize::MAX),
    (GuestAddress(0), 1),
];

assert!(matches!(
    memfd_backed(&regions, false, HugePageConfig::None),
    Err(MemoryError::OffsetTooLarge)
));
```

**Review by zulinx86 — CHANGES_REQUESTED** [review]
Thank you for contributing and sorry for being late to review.

The DCO check fails due to mismatch between Signed-off-by and Author: https://github.com/firecracker-microvm/firecracker/pull/6140/checks?check_run_id=99761771917
```text
Author:
Daniel Alejandro Dominguez Diaz <danielfromarg@gmail.com>

Signed-off-by:
Daniel <danielfromarg@gmail.com>
```

Could you update the Signed-off-by line?
Thanks,

File: src/vmm/src/vstate/memory.rs (hunk 1 above)
Comment (dan1d) [reply src/vmm/src/vstate/memory.rs:922]: Good call — added the test, pretty much as you suggested. Thanks!

**Comment by dan1d** [comment]
Fixed the Signed-off-by line and rebased on latest main — DCO passes now. Thanks for the review!

**Review by zulinx86 — APPROVED** [review]
Thank you for addressing the feedback! Looks good to me :)

**Review by ilstam — APPROVED** [review]
